resolve CData section too big on Export grid issue 24311

gaurangpadhiyar007 opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->

### Fixed Issues 
1. magento/magento2#24311: "CData section too big" error while accessing configurable product in backend 

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. At export history grid number of files more (>  20000)

### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

gaurangpadhiyar007 committed:
resolve CData section too big on Export grid issue 24311

Stepa4man commented:
@magento run all tests

gaurangpadhiyar007 commented:
@magento area adminhtml

gaurangpadhiyar007 commented:
@magento-cicd2 @magento-admin run area adminhtml

@magento-cicd2 @magento-admin run all tests

gaurangpadhiyar007 commented:
@magento run area adminhtml
@magento run all tests

rodrigowebjump commented:
Hi @gaurangpadhiyar007 Please, 
Thanks for your contribution.
I kindly ask you to add an appropriate description and tests scenarios to your PR.

Please, do not forget to review the failing tests.

Thanks
Thanks

gaurangpadhiyar007 committed:
Invalid arguments solved

Magento XML schema validation issue solved

gaurangpadhiyar007 committed:
Pagination Issue and basename warning solved

Pagination issue resolved
function basename warning resolved

gaurangpadhiyar007 reviewed (commented):
Issue of merging run test solved

gaurangpadhiyar007 committed:
Merge issue due to params resolved

$meta parameter is not in order
@param is not found for one or more params in method annotation
$fileIO parameter is missing in method annotation

gaurangpadhiyar007 committed:
Merging issue of ui xml validation resolved

This node should migrate to "settings/sticky"
This node should migrate to "settings/selectProvider"
This node should migrate to "settings/storageConfig"

gaurangpadhiyar007 committed:
Update export_grid.xml

gaurangpadhiyar007 committed:
Update export_grid.xml

gaurangpadhiyar007 committed:
Avoid unused private fields such as $fileIO

gaurangpadhiyar007 commented:
@rodrigowebjump
 I have made all changes as per given errors but as I solve them new un-neccessary errors are generating by Automated testing tool, Please test the commit on Magento instance.

sidolov commented on app/code/Magento/ImportExport/Ui/DataProvider/ExportFileDataProvider.php:
```diff
@@ -51,11 +60,14 @@ public function __construct(
         \Magento\Framework\Api\FilterBuilder $filterBuilder,
         DriverInterface $file,
         Filesystem $filesystem,
+        File $fileIO,
```
Please, inject new dependencies according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

sidolov commented:
@gaurangpadhiyar007  I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!
